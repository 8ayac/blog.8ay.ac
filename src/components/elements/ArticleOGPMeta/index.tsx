import { config } from '@/site.config';
import { getLastModifiedDate } from '@/src/shared/utils';
import { Article } from '@/src/types';
import React from 'react';

export const ArticleOGPMeta: React.FC<{
  article: Article;
}> = ({ article }) => (
  <>
    <meta
      property="article:published_time"
      content={new Date(article.attributes.publishedAt).toISOString()}
    />
    <meta
      property="article:modified_time"
      content={getLastModifiedDate(article).toISOString()}
    />
    <meta property="article:author" content={config.site.maintainer} />
    {[...new Set(article.attributes.tags)].map((tag) => (
      <meta key={tag} property="article:tag" content={tag} />
    ))}
  </>
);
