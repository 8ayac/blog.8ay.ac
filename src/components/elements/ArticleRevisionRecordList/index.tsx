import { REPOSITORY_URL } from '@/src/constants/site';
import { ArticleChangeLog } from '@/src/types';
import styled from '@emotion/styled';
import React from 'react';
import urljoin from 'url-join';

const S = {
  Details: styled.details``,

  Summary: styled.summary`
    margin-bottom: 0.75rem;
    font-weight: 600;
    cursor: pointer;
    outline: none;
  `,

  LogList: styled.ul`
    padding-left: 0;
    margin: 0;
  `,

  LogListItem: styled.li`
    margin-bottom: 0.75rem;
    list-style: none;
  `,

  UpdateDateTime: styled.time``,

  UpdateDescriptionSpan: styled.span`
    font-weight: bolder;
  `,

  ToGitHubCommitAnchor: styled.a``,
};

const removeGitCommitPrefix = (commitMessage: string): string => {
  return commitMessage.replace(/^:.*?:\s+/, '');
};

const getGitCommitUrl = (commitId: string): string => {
  return urljoin(REPOSITORY_URL, 'commit', commitId);
};

const LogList: React.FC<{ log: ArticleChangeLog[] }> = ({ log }) => {
  return (
    <S.LogList>
      {log.map((record) => (
        <S.LogListItem key={record.id}>
          <S.UpdateDateTime>
            {new Date(record.date).toISOString()}
          </S.UpdateDateTime>{' '}
          <S.UpdateDescriptionSpan>
            {removeGitCommitPrefix(record.description)}
          </S.UpdateDescriptionSpan>{' '}
          (
          <S.ToGitHubCommitAnchor
            href={getGitCommitUrl(record.id)}
            target="_blank"
            rel="noopener"
          >
            #{record.id}
          </S.ToGitHubCommitAnchor>{' '}
          by {record.author})
        </S.LogListItem>
      ))}
    </S.LogList>
  );
};

export const ArticleRevisionRecordList: React.FC<{
  log: ArticleChangeLog[];
}> = ({ log }) => {
  log.sort((a, b) => new Date(a.date).getTime() - new Date(b.date).getTime());

  if (log.length < 1) return <></>;

  return (
    <>
      <S.Details open>
        <S.Summary>Revision History</S.Summary>
        <LogList log={log} />
      </S.Details>
    </>
  );
};
