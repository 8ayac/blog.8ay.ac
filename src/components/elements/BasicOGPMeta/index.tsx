import { config } from '@/site.config';
import React from 'react';
import urljoin from 'url-join';

export const BasicOGPMeta: React.FC<{
  ogTitle: string;
  ogType: string;
  ogDescription: string;
  ogImageUrl?: string;
  ogUrl: string;
}> = ({ ogTitle, ogDescription, ogImageUrl, ogUrl, ogType }) => (
  <>
    <meta property="og:title" content={ogTitle} />
    <meta property="og:type" content={ogType} />
    <meta property="og:url" content={ogUrl} />
    <meta
      property="og:image"
      content={ogImageUrl || urljoin(config.site.rootUrl, 'og.png')}
    />
    <meta property="og:site_name" content={config.site.title} />
    {!!ogDescription && (
      <meta property="og:description" content={ogDescription} />
    )}
  </>
);
