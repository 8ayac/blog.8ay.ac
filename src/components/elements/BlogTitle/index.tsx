import styled from '@emotion/styled';
import Link from 'next/link';
import React from 'react';

const S = {
  LogoWrapperAnchor: styled.a`
    color: inherit;
    text-decoration: none;
    cursor: pointer;
    user-select: none;

    &:visited {
      color: inherit;
    }

    &:hover {
      color: ${(props) =>
        props.theme.color.blogTitle.logoWrapperAnchor.textOnHover};
    }
  `,

  TitleSpan: styled.span`
    font-size: 2.2rem;
  `,

  SubtitleSpan: styled.span`
    margin-left: 0.5rem;
    font-size: 1.7rem;
  `,
};

const Logo: React.FC<{ title: string; subtitle: string; color?: string }> = ({
  title,
  subtitle,
}) => (
  <>
    <S.TitleSpan>{title}</S.TitleSpan>
    <S.SubtitleSpan>{subtitle}</S.SubtitleSpan>
  </>
);

export const BlogTitle: React.FC<{
  title: string;
  subtitle: string;
  linkToTopPage?: boolean;
}> = ({ title, subtitle, linkToTopPage = true }) => (
  <>
    {linkToTopPage ? (
      <Link href="/" passHref>
        <S.LogoWrapperAnchor>
          <Logo title={title} subtitle={subtitle} />
        </S.LogoWrapperAnchor>
      </Link>
    ) : (
      <Logo title={title} subtitle={subtitle} />
    )}
  </>
);
