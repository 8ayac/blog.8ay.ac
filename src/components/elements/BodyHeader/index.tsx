import { theme } from '@/src/constants/theme';
import styled from '@emotion/styled';
import Link from 'next/link';
import React, { ReactNode } from 'react';

export const S = {
  WrapperHeader: styled.header`
    margin-bottom: 3rem;
    border-bottom: 0.25rem solid ${theme.color.border.primaryLight};
  `,

  TitleH1: styled.h1`
    margin: 0 0 1rem;
    font-size: 2.16em;
  `,

  TitleAnchor: styled.a`
    margin: 0;
    color: ${theme.color.text.primary};
    text-decoration: none;
    filter: drop-shadow(2px 2px 1px ${theme.color.green.light2});
  `,

  BeforeTitleDiv: styled.div`
    margin-bottom: 0.3rem;
  `,

  AfterTitleDiv: styled.div`
    margin-top: 0.3rem;
  `,
};

export const BodyHeader: React.FC<{
  title: string;
  linkTo?: string;
  beforeTitle?: ReactNode;
  afterTitle?: ReactNode;
}> = ({ title, linkTo, beforeTitle, afterTitle }) => (
  <>
    <S.WrapperHeader>
      {beforeTitle && <S.BeforeTitleDiv>{beforeTitle}</S.BeforeTitleDiv>}
      {linkTo ? (
        <S.TitleH1>
          <Link href={linkTo} passHref>
            <S.TitleAnchor>{title}</S.TitleAnchor>
          </Link>
        </S.TitleH1>
      ) : (
        <S.TitleH1>{title}</S.TitleH1>
      )}
      {afterTitle && <S.AfterTitleDiv>{afterTitle}</S.AfterTitleDiv>}
    </S.WrapperHeader>
  </>
);
