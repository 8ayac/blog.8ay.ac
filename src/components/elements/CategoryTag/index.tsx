import { theme } from '@/src/constants/theme';
import { getTagsPagePath } from '@/src/shared/utils';
import styled from '@emotion/styled';
import Link from 'next/link';
import React from 'react';

const S = {
  Anchor: styled.a`
    padding: 3% 1rem 4%;
    margin: 0;
    font-size: 0.8em;
    font-weight: 600;
    color: white;
    text-decoration: none;
    background-color: ${theme.color.blue.dark};
    border-radius: 3px;

    &:hover {
      color: white;
      background-color: ${theme.color.blue.base};
    }
  `,
};

export const CategoryTag: React.FC<{ name: string }> = ({ name }) => (
  <>
    <Link href={getTagsPagePath(name)} passHref>
      <S.Anchor>{name}</S.Anchor>
    </Link>
  </>
);
