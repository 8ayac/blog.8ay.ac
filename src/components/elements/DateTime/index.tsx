import styled from '@emotion/styled';
import React from 'react';

const S = {
  ComponentWrapperDiv: styled.div`
    display: flex;
    flex-flow: row wrap;
  `,

  DescriptionSpan: styled.span`
    margin-right: 0.5rem;
    font-weight: 600;
  `,

  ISODateTime: styled.time`
    font-weight: bolder;
    color: ${(props) => props.theme.color.dateTime.isoDateTime.text};
  `,
};

export const DateTime: React.FC<{
  date: Date | string;
  description?: string;
}> = ({ date, description }) => (
  <>
    <S.ComponentWrapperDiv>
      {description && (
        <S.DescriptionSpan>{`${description}:`}</S.DescriptionSpan>
      )}
      <S.ISODateTime>{new Date(date).toISOString()}</S.ISODateTime>
    </S.ComponentWrapperDiv>
  </>
);
