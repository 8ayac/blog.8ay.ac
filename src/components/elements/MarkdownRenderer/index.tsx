import { theme } from '@/src/constants/theme';
import styled from '@emotion/styled';
import Prism from 'prismjs';
import React, { useEffect } from 'react';
import ReactMarkdown from 'react-markdown';
import remarkGfm from 'remark-gfm';

const S = {
  ConvertedMdBody: styled.div`
    font-size: 1.6rem;
    overflow-wrap: break-word;

    h1,
    h2 {
      padding-bottom: 0.3em;
      margin-top: 2em;
      border-bottom: 1px solid ${theme.color.border.primaryLight};
    }

    h3,
    h4,
    h5,
    h6 {
      margin-top: 1em;
    }

    a {
      font-weight: 600;
    }

    code,
    pre {
      font-family: SFMono-Regular, Consolas, Liberation Mono, Menlo, monospace;
      word-break: normal;
    }

    pre {
      white-space: pre-wrap;
    }

    /* This rule is for the inline code block. */
    code:not([class]) {
      padding: 0.2em 0.4em;
      margin: 0;
      font-size: 0.85em;
      background-color: hsla(0, 0%, 92.5%);
      border-radius: 3px;
    }

    blockquote,
    details,
    dl,
    ol,
    p,
    pre,
    table,
    ul {
      margin-top: 0;
      margin-bottom: 1.8rem;
    }

    ol ol,
    ul ol {
      list-style-type: lower-roman;
    }

    li {
      > p {
        margin-top: 1.6rem;
      }

      + li {
        margin-top: 0.25em;
      }
    }

    blockquote {
      padding: 0 1em;
      margin-left: 0;
      color: ${theme.color.text.primaryLight};
      border-left: 0.25em solid ${theme.color.blue.light3};

      :first-of-type {
        margin-top: 0;
      }

      :last-of-type {
        margin-bottom: 0;
      }
    }

    /* Below is for the plugin of Prism 'Show Language' */
    div.code-toolbar > .toolbar {
      top: 0.3em;
      right: 0.3em;
      transition: none;

      span {
        font-size: 0.85em;
        border-radius: 10px;
      }
    }
  `,
};

export const MarkdownRender: React.FC<{ content: string }> = ({ content }) => {
  useEffect(() => {
    Prism.highlightAll();
  });

  return (
    <>
      <S.ConvertedMdBody>
        <ReactMarkdown source={content} plugins={[remarkGfm]} />
      </S.ConvertedMdBody>
    </>
  );
};
