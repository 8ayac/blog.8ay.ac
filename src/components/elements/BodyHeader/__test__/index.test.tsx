import { BodyHeader } from '@/src/components/elements/BodyHeader';
import { theme } from '@/src/constants/theme';
import { mount, shallow } from 'enzyme';
import React from 'react';

describe('BodyHeader', () => {
  const testTitle = 'Test Title';
  const testLinkTo = '/';
  const testBeforeTitle = <div>BEFORE</div>;
  const testAfterTitle = <div>AFTER</div>;

  describe('is rendered correctly', () => {
    describe('to match the snapshot', () => {
      test('when linkTo is not given', () => {
        const wrapper = mount(
          <BodyHeader
            title={testTitle}
            beforeTitle={testBeforeTitle}
            afterTitle={testAfterTitle}
          />
        );
        expect(wrapper).toMatchSnapshot();
      });

      test('when linkTo is given', () => {
        const wrapper = mount(
          <BodyHeader
            title={testTitle}
            linkTo={testLinkTo}
            beforeTitle={testBeforeTitle}
            afterTitle={testAfterTitle}
          />
        );
        expect(wrapper).toMatchSnapshot();
      });
    });

    describe('under certain conditions', () => {
      test('linkTo: passed', () => {
        const wrapper = shallow(
          <BodyHeader
            title={testTitle}
            linkTo={testLinkTo}
            beforeTitle={testBeforeTitle}
            afterTitle={testAfterTitle}
          />
        );
        expect(wrapper.find('Link')).toHaveLength(1);
      });

      test('linkTo: NOT passed', () => {
        const wrapper = shallow(
          <BodyHeader
            title={testTitle}
            beforeTitle={testBeforeTitle}
            afterTitle={testAfterTitle}
          />
        );
        expect(wrapper.find('Link')).toHaveLength(0);
      });

      test('title: passed, beforeTitle: passed, afterTitle: passed', () => {
        const wrapper = shallow(
          <BodyHeader
            title={testTitle}
            beforeTitle={testBeforeTitle}
            afterTitle={testAfterTitle}
          />
        );

        expect(wrapper.find('TitleH1')).toHaveLength(1);
        expect(wrapper.find('BeforeTitleDiv')).toHaveLength(1);
        expect(wrapper.find('AfterTitleDiv')).toHaveLength(1);
      });

      test('title: passed, beforeTitle: passed, afterTitle: NOT passed', () => {
        const wrapper = shallow(
          <BodyHeader title={testTitle} beforeTitle={testBeforeTitle} />
        );

        expect(wrapper.find('TitleH1')).toHaveLength(1);
        expect(wrapper.find('BeforeTitleDiv')).toHaveLength(1);
        expect(wrapper.find('AfterTitleDiv')).toHaveLength(0);
      });

      test('title: passed, beforeTitle: NOT passed, afterTitle: passed', () => {
        const wrapper = shallow(
          <BodyHeader title={testTitle} afterTitle={testBeforeTitle} />
        );

        expect(wrapper.find('TitleH1')).toHaveLength(1);
        expect(wrapper.find('BeforeTitleDiv')).toHaveLength(0);
        expect(wrapper.find('AfterTitleDiv')).toHaveLength(1);
      });

      test('title: passed, beforeTitle: NOT passed, afterTitle: NOT passed', () => {
        const wrapper = shallow(<BodyHeader title={testTitle} />);

        expect(wrapper.find('TitleH1')).toHaveLength(1);
        expect(wrapper.find('BeforeTitleDiv')).toHaveLength(0);
        expect(wrapper.find('AfterTitleDiv')).toHaveLength(0);
      });
    });
  });

  describe('has proper style rules', () => {
    const wrapper = mount(
      <BodyHeader
        title={testTitle}
        linkTo={testLinkTo}
        beforeTitle={testBeforeTitle}
        afterTitle={testAfterTitle}
      />
    );

    test('in WrapperHeader', () => {
      expect(wrapper.find('WrapperHeader')).toHaveStyleRule(
        'margin-bottom',
        '3rem'
      );
      expect(wrapper.find('WrapperHeader')).toHaveStyleRule(
        'border-bottom',
        `0.25rem solid ${theme.color.border.primaryLight}`
      );
    });

    test('in TitleH1', () => {
      expect(wrapper.find('TitleH1')).toHaveStyleRule('font-size', '2.16em');
      expect(wrapper.find('TitleH1')).toHaveStyleRule('margin', '0 0 1rem');
    });

    test('in TitleAnchor', () => {
      expect(wrapper.find('TitleAnchor')).toHaveStyleRule('margin', '0');
      expect(wrapper.find('TitleAnchor')).toHaveStyleRule(
        'color',
        `${theme.color.text.primary}`
      );
      expect(wrapper.find('TitleAnchor')).toHaveStyleRule(
        'text-decoration',
        'none'
      );
    });

    test('in BeforeTitleDiv', () => {
      expect(wrapper.find('BeforeTitleDiv')).toHaveStyleRule(
        'margin-bottom',
        '0.3rem'
      );
    });

    test('in AfterTitleDiv:', () => {
      expect(wrapper.find('AfterTitleDiv')).toHaveStyleRule(
        'margin-top',
        '0.3rem'
      );
    });
  });
});
