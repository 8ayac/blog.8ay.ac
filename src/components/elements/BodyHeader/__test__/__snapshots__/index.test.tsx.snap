// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BodyHeader is rendered correctly to match the snapshot when linkTo is given 1`] = `
.emotion-0 {
  margin-bottom: 3rem;
  border-bottom: 0.25rem solid hsla(0, 0%, 75%);
}

.emotion-2 {
  margin-bottom: 0.3rem;
}

.emotion-4 {
  margin: 0 0 1rem;
  font-size: 2.16em;
}

.emotion-6 {
  margin: 0;
  color: hsla(0, 0%, 15%);
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-filter: drop-shadow(
        2px 2px 1px hsla(148, 72%, 80%)
      );
  filter: drop-shadow(
        2px 2px 1px hsla(148, 72%, 80%)
      );
}

.emotion-6:visited {
  color: hsla(0, 0%, 15%);
}

.emotion-8 {
  margin-top: 0.3rem;
}

<ThemeProvider
  theme={
    Object {
      "color": Object {
        "articleShareButtons": Object {
          "buttonWrapperDiv": Object {
            "shadow": "hsla(212, 72%, 80%)",
          },
        },
        "blogTitle": Object {
          "logoWrapperAnchor": Object {
            "textOnHover": "hsla(148, 72%, 80%)",
          },
        },
        "bodyHeader": Object {
          "titleAnchor": Object {
            "shadow": "hsla(148, 72%, 80%)",
          },
        },
        "categoryTag": Object {
          "anchor": Object {
            "bg": "hsla(212, 72%, 50%)",
            "bgOnHover": "hsla(212, 72%, 60%)",
            "shadow": "hsla(212, 72%, 80%)",
            "text": "white",
          },
        },
        "common": Object {
          "border": Object {
            "primary": "hsla(0, 0%, 50%)",
            "primaryLight": "hsla(0, 0%, 75%)",
          },
          "text": Object {
            "accent": "hsla(148, 72%, 50%)",
            "primary": "hsla(0, 0%, 15%)",
            "primaryLight": "hsla(0, 0%, 40%)",
            "secondary": "hsla(212, 72%, 40%)",
          },
        },
        "dateTime": Object {
          "isoDateTime": Object {
            "text": "hsla(212, 72%, 40%)",
          },
        },
        "global": Object {
          "body": Object {
            "bg": "hsla(60, 72%, 60%)",
          },
          "table": Object {
            "tr": Object {
              "bg2n": "hsla(148, 72%, 90%)",
            },
          },
        },
        "markdownRenderer": Object {
          "convertedMdBody": Object {
            "blockquote": Object {
              "border": "hsla(212, 72%, 90%)",
            },
          },
        },
        "siteHeader": Object {
          "header": Object {
            "bg": "hsla(148, 72%, 30%)",
            "shadow": "hsla(148, 72%, 20%)",
          },
          "innerWrapperFlexContainerDiv": Object {
            "shadow": "hsla(148, 72%, 20%)",
          },
        },
        "socialLinkWithIcon": Object {
          "socialLinkAnchor": Object {
            "textOnHover": "hsla(60, 72%, 60%)",
          },
        },
      },
    }
  }
>
  <BodyHeader
    afterTitle={
      <div>
        AFTER
      </div>
    }
    beforeTitle={
      <div>
        BEFORE
      </div>
    }
    linkTo="/"
    title="Test Article Title 1"
  >
    <WrapperHeader>
      <header
        className="emotion-0 emotion-1"
      >
        <BeforeTitleDiv>
          <div
            className="emotion-2 emotion-3"
          >
            <div>
              BEFORE
            </div>
          </div>
        </BeforeTitleDiv>
        <TitleH1>
          <h1
            className="emotion-4 emotion-5"
          >
            <Link
              href="/"
              passHref={true}
            >
              <TitleAnchor
                href="/"
                onClick={[Function]}
                onMouseEnter={[Function]}
              >
                <a
                  className="emotion-6 emotion-7"
                  href="/"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                >
                  Test Article Title 1
                </a>
              </TitleAnchor>
            </Link>
          </h1>
        </TitleH1>
        <AfterTitleDiv>
          <div
            className="emotion-8 emotion-9"
          >
            <div>
              AFTER
            </div>
          </div>
        </AfterTitleDiv>
      </header>
    </WrapperHeader>
  </BodyHeader>
</ThemeProvider>
`;

exports[`BodyHeader is rendered correctly to match the snapshot when linkTo is not given 1`] = `
.emotion-0 {
  margin-bottom: 3rem;
  border-bottom: 0.25rem solid hsla(0, 0%, 75%);
}

.emotion-2 {
  margin-bottom: 0.3rem;
}

.emotion-4 {
  margin: 0 0 1rem;
  font-size: 2.16em;
}

.emotion-6 {
  margin-top: 0.3rem;
}

<ThemeProvider
  theme={
    Object {
      "color": Object {
        "articleShareButtons": Object {
          "buttonWrapperDiv": Object {
            "shadow": "hsla(212, 72%, 80%)",
          },
        },
        "blogTitle": Object {
          "logoWrapperAnchor": Object {
            "textOnHover": "hsla(148, 72%, 80%)",
          },
        },
        "bodyHeader": Object {
          "titleAnchor": Object {
            "shadow": "hsla(148, 72%, 80%)",
          },
        },
        "categoryTag": Object {
          "anchor": Object {
            "bg": "hsla(212, 72%, 50%)",
            "bgOnHover": "hsla(212, 72%, 60%)",
            "shadow": "hsla(212, 72%, 80%)",
            "text": "white",
          },
        },
        "common": Object {
          "border": Object {
            "primary": "hsla(0, 0%, 50%)",
            "primaryLight": "hsla(0, 0%, 75%)",
          },
          "text": Object {
            "accent": "hsla(148, 72%, 50%)",
            "primary": "hsla(0, 0%, 15%)",
            "primaryLight": "hsla(0, 0%, 40%)",
            "secondary": "hsla(212, 72%, 40%)",
          },
        },
        "dateTime": Object {
          "isoDateTime": Object {
            "text": "hsla(212, 72%, 40%)",
          },
        },
        "global": Object {
          "body": Object {
            "bg": "hsla(60, 72%, 60%)",
          },
          "table": Object {
            "tr": Object {
              "bg2n": "hsla(148, 72%, 90%)",
            },
          },
        },
        "markdownRenderer": Object {
          "convertedMdBody": Object {
            "blockquote": Object {
              "border": "hsla(212, 72%, 90%)",
            },
          },
        },
        "siteHeader": Object {
          "header": Object {
            "bg": "hsla(148, 72%, 30%)",
            "shadow": "hsla(148, 72%, 20%)",
          },
          "innerWrapperFlexContainerDiv": Object {
            "shadow": "hsla(148, 72%, 20%)",
          },
        },
        "socialLinkWithIcon": Object {
          "socialLinkAnchor": Object {
            "textOnHover": "hsla(60, 72%, 60%)",
          },
        },
      },
    }
  }
>
  <BodyHeader
    afterTitle={
      <div>
        AFTER
      </div>
    }
    beforeTitle={
      <div>
        BEFORE
      </div>
    }
    title="Test Article Title 1"
  >
    <WrapperHeader>
      <header
        className="emotion-0 emotion-1"
      >
        <BeforeTitleDiv>
          <div
            className="emotion-2 emotion-3"
          >
            <div>
              BEFORE
            </div>
          </div>
        </BeforeTitleDiv>
        <TitleH1>
          <h1
            className="emotion-4 emotion-5"
          >
            Test Article Title 1
          </h1>
        </TitleH1>
        <AfterTitleDiv>
          <div
            className="emotion-6 emotion-7"
          >
            <div>
              AFTER
            </div>
          </div>
        </AfterTitleDiv>
      </header>
    </WrapperHeader>
  </BodyHeader>
</ThemeProvider>
`;
